plugins/vli/fu-vli-usbhub-i2c-device.c

```diff
@@ -246,6 +247,16 @@ fu_vli_usbhub_i2c_device_write_firmware (FuDevice *device,
 		fu_device_set_progress_full (device, (gsize) j, (gsize) records->len);
 	}
 
+	/* the device automatically reboots */
+	fu_device_set_status (device, FWUPD_STATUS_DEVICE_RESTART);
+	fu_device_set_progress (device, 0);
+
+	/* this is unusual, but the MSP device reboot takes down the entire hub
+	 * for ~60 seconds and we don't want the parent device removing us */
+	root = fu_device_get_root (device);
```

Comment:
do you maybe want an assertion that root != self at the top of this function?

Reply:
Not sure. I don;t think it's ever possible.

Reply:
If parent wasn't set somehow, that seems like a bad programming error, which is why I thought an assertion made sense.